src/panels/config/script/ha-script-editor.ts

```diff
@@ -786,24 +802,46 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
     }
     const id = this.scriptId || this._entityId || Date.now();
 
+    this._saving = true;
     try {
       await this.hass!.callApi(
         "POST",
         "config/script/config/" + id,
         this._config
       );
+
+      if (this._entityRegistryUpdate !== undefined) {
+        // wait for new script to appear in entity registry
+        if (!this.scriptId) {
+          await new Promise<void>((resolve, _reject) => {
+            setTimeout(resolve, 3000);
+          });
+        }
```

Comment:
Not a fan of this...

Reply:
Also don't have a quick solution, a nicer option could be a promise that is resolved in the `willUpdate` handler when the entity registry is updated

Reply:
We need to delay the `updateEntityRegistryEntry` nevertheless, the `subscribeEntityRegistryUpdates` is debounced for 500ms. If we remove the `setTimeout` completely now, we would not get the updated entity entry, as it is debounced, which would result in a missing label/category on the list page until a complete reload occurs.

We could now directly subscribe the `entity_registry_updated` in the `ha-*-editor.ts`, bypassing the debounce, and unsubscribe after we received the confirmation that the entity has been created. What do you think?

Reply:
Nvm, we wouldn't get the second `entity_registry_updated` either, as the debounce would still be called, as other components are still subscribed.